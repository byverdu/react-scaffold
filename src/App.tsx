import React from 'react';
import Image from 'Components/Image';
import './App.css';
import { hot } from 'react-hot-loader';

const App: React.FC = () => {
  return (
    <div className="App">
      <header className="App-header">
        <Image />
        <p>
          Edit <code>xoxo</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
};

export default hot(module)(App);
