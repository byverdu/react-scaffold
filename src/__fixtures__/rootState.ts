import { TodoState, VisibilityFilterState } from 'Models/Todo';
import { RootState } from 'Core/reducers';
import { VisibilityFilterEnum } from 'Models/Enums';
import { Map } from 'immutable';

const initialSet = {
  1234: { id: '1234', text: 'todo task', done: false }
};

const todos: TodoState = {
  todos: Map(initialSet)
};

const visibilityFilter: VisibilityFilterState = {
  activeFilter: VisibilityFilterEnum.SHOW_ALL
};

const rootState: RootState = {
  todos,
  visibilityFilter
};

export default rootState;
