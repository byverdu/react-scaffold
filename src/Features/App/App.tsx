import Image from 'Components/Image';
import React from 'react';
import { hot } from 'react-hot-loader';
import styles from 'Theme/App.module.scss';
import 'Theme/App.scss';

const App: React.FC = () => {
  return (
    <div className="App">
      <header className="App-header">
        <Image />
        <p>
          Edit <code>xoxo</code> and save to reload.
        </p>
        <a
          className={styles.link}
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
};

export default hot(module)(App);
