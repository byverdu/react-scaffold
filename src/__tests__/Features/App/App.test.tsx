import React from 'react';
import ReactDOM from 'react-dom';
import App from 'Features/App/App';
import enzyme, { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import rootState from '__fixtures__/rootState';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore(rootState);

describe('<App />', () => {
  let wrapper;
  const component = () => (
    <Provider store={store}>
      <App />
    </Provider>
  );

  beforeEach(() => {
    wrapper = mount(component());
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component(), div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should create a snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render a TodosContainer', () => {
    expect(wrapper.find('TodosContainer')).toHaveLength(1);
  });
});
