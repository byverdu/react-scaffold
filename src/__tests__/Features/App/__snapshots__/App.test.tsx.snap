// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should create a snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(App)>
    <App
      getTodos={[Function]}
    >
      <div
        className="App"
      >
        <header
          className="App-header"
        >
          <Image>
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />
          </Image>
          <h1>
            Todo List App
          </h1>
        </header>
        <Connect(TodosContainer)>
          <TodosContainer
            addTodo={[Function]}
            checkHandler={[Function]}
            deleteTodo={[Function]}
            todos={
              Immutable.Map {
                "1234": Object {
                  "done": false,
                  "id": "1234",
                  "text": "todo task",
                },
              }
            }
            todosLength={1}
          >
            <AddTodo
              addTodo={[Function]}
            >
              <section>
                <h2>
                  Add a Todo
                </h2>
                <form
                  noValidate={true}
                >
                  <label
                    htmlFor="add-todo-field"
                  >
                    Todo text
                  </label>
                  <input
                    id="add-todo-field"
                    onChange={[Function]}
                    type="text"
                  />
                  <button
                    disabled={true}
                    onClick={[Function]}
                  >
                    Submit
                  </button>
                </form>
              </section>
            </AddTodo>
            <section>
              <h3>
                Showing 1 todo tasks
              </h3>
              <Connect(VisibilityFilter)>
                <VisibilityFilter
                  activeFilter="SHOW_ALL"
                  filterHandler={[Function]}
                >
                  <section>
                    <label
                      htmlFor="SHOW_ALL"
                      key="SHOW_ALL"
                    >
                      Show All
                      <input
                        checked={true}
                        id="SHOW_ALL"
                        name="filter-type"
                        onChange={[Function]}
                        type="radio"
                        value="SHOW_ALL"
                      />
                    </label>
                    <label
                      htmlFor="SHOW_DONE"
                      key="SHOW_DONE"
                    >
                      Show Done
                      <input
                        checked={false}
                        id="SHOW_DONE"
                        name="filter-type"
                        onChange={[Function]}
                        type="radio"
                        value="SHOW_DONE"
                      />
                    </label>
                    <label
                      htmlFor="SHOW_TODO"
                      key="SHOW_TODO"
                    >
                      Show Todo
                      <input
                        checked={false}
                        id="SHOW_TODO"
                        name="filter-type"
                        onChange={[Function]}
                        type="radio"
                        value="SHOW_TODO"
                      />
                    </label>
                  </section>
                </VisibilityFilter>
              </Connect(VisibilityFilter)>
              <Todos
                checkHandler={[Function]}
                deleteTodo={[Function]}
                todos={
                  Immutable.Map {
                    "1234": Object {
                      "done": false,
                      "id": "1234",
                      "text": "todo task",
                    },
                  }
                }
              >
                <ul>
                  <TodoItem
                    checkHandler={[Function]}
                    deleteTodo={[Function]}
                    key="1234"
                    todo={
                      Object {
                        "done": false,
                        "id": "1234",
                        "text": "todo task",
                      }
                    }
                  >
                    <li
                      className="task-todo"
                      style={
                        Object {
                          "textDecoration": "none",
                        }
                      }
                    >
                      <label
                        data-id="1234"
                        htmlFor="list-item-1234"
                      >
                        <input
                          checked={false}
                          id="list-item-1234"
                          onChange={[Function]}
                          type="checkbox"
                        />
                        todo task
                      </label>
                      <button
                        onClick={[Function]}
                      >
                        Delete
                      </button>
                    </li>
                  </TodoItem>
                </ul>
              </Todos>
            </section>
          </TodosContainer>
        </Connect(TodosContainer)>
      </div>
    </App>
  </Connect(App)>
</Provider>
`;
